internal/translation/dbuser/dbuser_test.go

```diff
@@ -40,7 +43,149 @@ func TestFailedNewAtlasDatabaseUsersService(t *testing.T) {
 	require.Nil(t, users)
 	require.ErrorIs(t, err, expectedErr)
 }
+func TestAtlasUsersGet(t *testing.T) {
+	ctx := context.Background()
+	projectID := "project-id"
+	db := "database"
+	username := "test-user"
+
+	tests := []struct {
+		name         string
+		setupMock    func(mockUsersAPI *mockadmin.DatabaseUsersApi)
+		expectedUser *admin.CloudDatabaseUser // Replace with actual user type
+		expectedErr  string
+	}{
+		{
+			name: "User found",
+			setupMock: func(mockUsersAPI *mockadmin.DatabaseUsersApi) {
+				expectedUser := &admin.CloudDatabaseUser{DatabaseName: db, GroupId: projectID, Username: username}
+				mockUsersAPI.EXPECT().GetDatabaseUser(ctx, projectID, db, username).Return(
+					admin.GetDatabaseUserApiRequest{ApiService: mockUsersAPI})
+
+				mockUsersAPI.EXPECT().GetDatabaseUserExecute(admin.GetDatabaseUserApiRequest{ApiService: mockUsersAPI}).Return(
+					expectedUser, &http.Response{StatusCode: http.StatusOK}, nil)
+			},
+			expectedUser: &admin.CloudDatabaseUser{DatabaseName: db, GroupId: projectID, Username: username},
+			expectedErr:  "",
+		},
+		{
+			name: "User not found",
+			setupMock: func(mockUsersAPI *mockadmin.DatabaseUsersApi) {
+				mockUsersAPI.EXPECT().GetDatabaseUser(ctx, projectID, db, username).Return(
+					admin.GetDatabaseUserApiRequest{ApiService: mockUsersAPI})
+
+				notFoundErr := &admin.GenericOpenAPIError{}
+				notFoundErr.SetModel(admin.ApiError{ErrorCode: pointer.MakePtr("USERNAME_NOT_FOUND")})
+				mockUsersAPI.EXPECT().GetDatabaseUserExecute(admin.GetDatabaseUserApiRequest{ApiService: mockUsersAPI}).Return(
+					nil, &http.Response{StatusCode: http.StatusNotFound}, notFoundErr)
+			},
+			expectedUser: nil,
+			expectedErr:  "database user not found\n",
+		},
+		{
+			name: "API error",
+			setupMock: func(mockUsersAPI *mockadmin.DatabaseUsersApi) {
+				mockUsersAPI.EXPECT().GetDatabaseUser(ctx, projectID, db, username).Return(
+					admin.GetDatabaseUserApiRequest{ApiService: mockUsersAPI})
+
+				internalServerError := &admin.GenericOpenAPIError{}
+				internalServerError.SetModel(admin.ApiError{ErrorCode: pointer.MakePtr("500")})
+				mockUsersAPI.EXPECT().GetDatabaseUserExecute(admin.GetDatabaseUserApiRequest{ApiService: mockUsersAPI}).Return(
+					nil, &http.Response{StatusCode: http.StatusInternalServerError}, fmt.Errorf("some problem"))
+			},
+			expectedUser: nil,
+			expectedErr:  fmt.Sprintf("failed to get database user %q: %s", username, "some problem"),
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			mockUsersAPI := mockadmin.NewDatabaseUsersApi(t)
+			tt.setupMock(mockUsersAPI)
+
+			dus := &AtlasUsers{
+				usersAPI: mockUsersAPI,
+			}
+			user, err := dus.Get(ctx, db, projectID, username)
+
+			if tt.expectedErr != "" {
+				require.Equal(t, tt.expectedErr, err.Error())
+			} else {
+				require.NoError(t, err)
+			}
+			kubUser, _ := fromAtlas(tt.expectedUser)
```

Comment:
make expectation be already in the you want to assert, this conversion implies logic you are not testing